tests/src/unit-element_access2.cpp

```diff
@@ -423,10 +423,33 @@ TEST_CASE_TEMPLATE("element access 2", Json, nlohmann::json, nlohmann::ordered_j
 
                     SECTION("array")
                     {
-                        Json j_nonobject(Json::value_t::array);
-                        const Json j_nonobject_const(Json::value_t::array);
-                        CHECK_THROWS_WITH_AS(j_nonobject.value("/foo"_json_pointer, 1), "[json.exception.type_error.306] cannot use value() with array", typename Json::type_error&);
-                        CHECK_THROWS_WITH_AS(j_nonobject_const.value("/foo"_json_pointer, 1), "[json.exception.type_error.306] cannot use value() with array", typename Json::type_error&);
+                        Json j_array = Json::array({j});
+                        const Json j_array_const = Json::array({j});
```

Comment:
Since this function isn't varied on const vs non-const object, is having the two sets of tests useful?  There is a different function for lvalue vs rvalue for the default, so that might be a better thing to test.

Reply:
That's a rabbit hole I was hoping to avoid (in this PR). There's more I'd want to change...

Reply:
Okay.